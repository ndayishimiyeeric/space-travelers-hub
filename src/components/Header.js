import React from 'react';
import { Link } from 'react-router-dom';
import styles from './styles/Header.module.css';
import logo from './images/logo.png';

function Header() {
  const [active, setActive] = React.useState({
    rocket: true,
    mission: false,
    profile: false,
  });

  const style = {
    borderBottom: '2px solid #344563',
  };

  const setRocket = () => {
    setActive({
      rocket: true,
      mission: false,
      profile: false,
    });
  };

  const setMission = () => {
    setActive({
      rocket: false,
      mission: true,
      profile: false,
    });
  };

  const setProfile = () => {
    setActive({
      rocket: false,
      mission: false,
      profile: true,
    });
  };

  return (
    <nav className={styles.navbar}>
      <div className={styles.logodiv}>
        <img className={styles.logo} src={logo} alt="logo" />
      </div>
      <ul className={styles.links}>
        <li>
          <Link
            className={styles.link}
            style={active.rocket ? style : null}
            to="/"
            onClick={setRocket}
          >
            Rockets
          </Link>
        </li>
        <li>
          <Link
            className={styles.link}
            style={active.mission ? style : null}
            to="/Missions"
            onClick={setMission}
          >
            Missions
          </Link>
        </li>
        <li>
          <Link
            className={styles.link}
            style={active.profile ? style : null}
            to="/Profile"
            onClick={setProfile}
          >
            My Profile
          </Link>
        </li>
      </ul>
    </nav>
  );
}

export default Header;
