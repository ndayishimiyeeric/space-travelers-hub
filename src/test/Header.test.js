import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router, Link } from 'react-router-dom';
import Header from '../components/Header';

describe('Header render correctly', () => {
  test('test Rockets Link', () => {
    const tree = renderer
      .create(<Router><Link to="/rockets">Rockets</Link></Router>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('test Missions Link', () => {
    const tree = renderer
      .create(<Router><Link to="/missions">Missions</Link></Router>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('test My Profile Link', () => {
    const tree = renderer
      .create(<Router><Link to="/profile">My Profile</Link></Router>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('test Header', () => {
    const tree = renderer
      .create(<Router><Header /></Router>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
