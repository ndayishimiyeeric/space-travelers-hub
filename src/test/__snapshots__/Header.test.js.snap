// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header render correctly test Header 1`] = `
Array [
  <nav
    className="navbar"
  >
    <div
      className="logodiv"
    >
      <img
        alt="logo"
        className="logo"
        src="logo.png"
      />
      <p
        className="logoP"
      >
        Space Travelers Hub
      </p>
    </div>
    <ul
      className="links"
    >
      <li>
        <a
          className="link"
          href="/"
          onClick={[Function]}
          style={
            Object {
              "borderBottom": "2px solid #344563",
            }
          }
        >
          Rockets
        </a>
      </li>
      <li>
        <a
          className="link"
          href="/Missions"
          onClick={[Function]}
          style={null}
        >
          Missions
        </a>
      </li>
      <li>
        <a
          className="link"
          href="/Profile"
          onClick={[Function]}
          style={null}
        >
          My Profile
        </a>
      </li>
    </ul>
  </nav>,
  <hr />,
]
`;

exports[`Header render correctly test Missions Link 1`] = `
<a
  href="/missions"
  onClick={[Function]}
>
  Missions
</a>
`;

exports[`Header render correctly test My Profile Link 1`] = `
<a
  href="/profile"
  onClick={[Function]}
>
  My Profile
</a>
`;

exports[`Header render correctly test Rockets Link 1`] = `
<a
  href="/rockets"
  onClick={[Function]}
>
  Rockets
</a>
`;
